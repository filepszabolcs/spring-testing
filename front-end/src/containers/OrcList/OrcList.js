import React, {Component} from 'react';
import axios from "axios/index";

import OrcListItem from '../../components/OrcListItem/OrcListItem';


class OrcList extends Component {
    state = {
        orcs: []
    };

    getOrcs = () => {
        axios.get('/api/orcs')
            .then(response => {
                this.setState({
                    orcs: response.data
                })
            })
            .catch(error => {
                console.log(error.response);
            });
    }

    componentDidMount() {
        this.getOrcs();
    }

    deleteOrcHandler = (id) => {
        axios.delete('/api/orcs/' + id)
            .then(response => {
                console.log(response);
                this.getOrcs();
            })
            .catch(error => {
                console.log(error.response);
            });
    }

    render() {
        let orcs;
        orcs = this.state.orcs.map(orc => {
            return (
                <OrcListItem
                    key={orc.id}
                    orc={orc}
                    deleteHandler={this.deleteOrcHandler}
                />
            );
        });


        return (
            <div>
                <h3>Orc list</h3>
                <table className="table table-striped table-bordered table-condensed table-hover">
                    <thead>
                    <tr>
                        <th>Id</th>
                        <th>Name</th>
                        <th>Race type</th>
                        <th>Kill count</th>
                        <th>Weapons</th>
                        <th></th>
                        <th></th>
                    </tr>
                    </thead>
                    <tbody>
                        {orcs}
                    </tbody>
                </table>
            </div>
        );
    }

};

export default OrcList;