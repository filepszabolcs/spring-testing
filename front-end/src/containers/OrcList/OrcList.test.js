import React from 'react'
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import OrcList from './OrcList';

configure({adapter: new Adapter()});

describe('<OrcList />', () => {

    let wrapper;

    beforeEach(() => {
        wrapper = shallow(
            <OrcList />
        )
    })

    it('has a table with 7 columns', () => {
        expect(wrapper.find('.table').first().find('th')).toHaveLength(7);
    })

    it('has title with "Orc list"', () => {
        expect(wrapper.find('h3').first().text()).toBe('Orc list');
    })

    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    })

})