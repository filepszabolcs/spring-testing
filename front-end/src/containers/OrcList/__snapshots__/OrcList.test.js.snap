// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrcList /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OrcList />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          Orc list
</h3>,
        <table
          className="table table-striped table-bordered table-condensed table-hover"
>
          <thead>
                    <tr>
                              <th>
                                        Id
                              </th>
                              <th>
                                        Name
                              </th>
                              <th>
                                        Race type
                              </th>
                              <th>
                                        Kill count
                              </th>
                              <th>
                                        Weapons
                              </th>
                              <th />
                              <th />
                    </tr>
          </thead>
          <tbody>
                    
          </tbody>
</table>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Orc list",
        },
        "ref": null,
        "rendered": "Orc list",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <thead>
              <tr>
                            <th>
                                          Id
                            </th>
                            <th>
                                          Name
                            </th>
                            <th>
                                          Race type
                            </th>
                            <th>
                                          Kill count
                            </th>
                            <th>
                                          Weapons
                            </th>
                            <th />
                            <th />
              </tr>
</thead>,
            <tbody>
              
</tbody>,
          ],
          "className": "table table-striped table-bordered table-condensed table-hover",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <tr>
                <th>
                                Id
                </th>
                <th>
                                Name
                </th>
                <th>
                                Race type
                </th>
                <th>
                                Kill count
                </th>
                <th>
                                Weapons
                </th>
                <th />
                <th />
</tr>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <th>
                    Id
</th>,
                  <th>
                    Name
</th>,
                  <th>
                    Race type
</th>,
                  <th>
                    Kill count
</th>,
                  <th>
                    Weapons
</th>,
                  <th />,
                  <th />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Id",
                  },
                  "ref": null,
                  "rendered": "Id",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Name",
                  },
                  "ref": null,
                  "rendered": "Name",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Race type",
                  },
                  "ref": null,
                  "rendered": "Race type",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Kill count",
                  },
                  "ref": null,
                  "rendered": "Kill count",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Weapons",
                  },
                  "ref": null,
                  "rendered": "Weapons",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "th",
                },
              ],
              "type": "tr",
            },
            "type": "thead",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [],
            },
            "ref": null,
            "rendered": Array [],
            "type": "tbody",
          },
        ],
        "type": "table",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            Orc list
</h3>,
          <table
            className="table table-striped table-bordered table-condensed table-hover"
>
            <thead>
                        <tr>
                                    <th>
                                                Id
                                    </th>
                                    <th>
                                                Name
                                    </th>
                                    <th>
                                                Race type
                                    </th>
                                    <th>
                                                Kill count
                                    </th>
                                    <th>
                                                Weapons
                                    </th>
                                    <th />
                                    <th />
                        </tr>
            </thead>
            <tbody>
                        
            </tbody>
</table>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Orc list",
          },
          "ref": null,
          "rendered": "Orc list",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <thead>
                <tr>
                                <th>
                                                Id
                                </th>
                                <th>
                                                Name
                                </th>
                                <th>
                                                Race type
                                </th>
                                <th>
                                                Kill count
                                </th>
                                <th>
                                                Weapons
                                </th>
                                <th />
                                <th />
                </tr>
</thead>,
              <tbody>
                
</tbody>,
            ],
            "className": "table table-striped table-bordered table-condensed table-hover",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <tr>
                  <th>
                                    Id
                  </th>
                  <th>
                                    Name
                  </th>
                  <th>
                                    Race type
                  </th>
                  <th>
                                    Kill count
                  </th>
                  <th>
                                    Weapons
                  </th>
                  <th />
                  <th />
</tr>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <th>
                      Id
</th>,
                    <th>
                      Name
</th>,
                    <th>
                      Race type
</th>,
                    <th>
                      Kill count
</th>,
                    <th>
                      Weapons
</th>,
                    <th />,
                    <th />,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Id",
                    },
                    "ref": null,
                    "rendered": "Id",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Name",
                    },
                    "ref": null,
                    "rendered": "Name",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Race type",
                    },
                    "ref": null,
                    "rendered": "Race type",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Kill count",
                    },
                    "ref": null,
                    "rendered": "Kill count",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Weapons",
                    },
                    "ref": null,
                    "rendered": "Weapons",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "th",
                  },
                ],
                "type": "tr",
              },
              "type": "thead",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [],
              },
              "ref": null,
              "rendered": Array [],
              "type": "tbody",
            },
          ],
          "type": "table",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
